### PR TITLE
Finished implementation for profile setting page

### PR DESCRIPTION
**Pull request by LTan-101104**
Finished implementing profile setting.

### CODE REVIEW

File: frontend/source/components/ProfileSettingComponent/ProfileSettingComponent.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    <label class="switch">
+                        <input type="checkbox" id = "noti_pref">
+                        <span class="slider round"></span>
+                    </label>
+                </div>
+            </div>
+            <div class="footer">
+                <button type="submit" class="submit-button">Save</button>
+            </div>
+        </form>
```
Comment (joshua041904) [inline frontend/source/components/ProfileSettingComponent/ProfileSettingComponent.js:82]: HTML looks good

File: frontend/source/components/ProfileSettingComponent/ProfileSettingComponent.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        //add validation for inputs, just an alart for wrong input would be enough
+        //add fetching events to fetch current data to input field
+        const username = this.#container.querySelector("#name")
+        const email = this.#container.querySelector("#email")
+        const primary_tz = this.#container.querySelector("#primary-time-zone")
+        const secondary_tz = this.#container.querySelector("#secondary-time-zone")
+        const email_noti = this.#container.querySelector("#noti_pref")
+        const submit_button = this.#container.querySelector(".submit-button")
+        const data = { username, email, primary_tz, secondary_tz, email_noti }
+        submit_button.addEventListener("click", () => this.#handleSubmitData(data))
```
Comment (joshua041904) [inline frontend/source/components/ProfileSettingComponent/ProfileSettingComponent.js:98]: Event listener looks good

File: frontend/source/eventhub/Events.js [hunk 3]
```diff
@@ -25,4 +25,8 @@ export const Events = {
   SwitchToAuthenticationView: 'SwitchToAuthenticationView',
   SwitchToFriendsListView: 'SwitchToFriendsListView',
   SwitchToDateFinalizationView: 'SwitchToDateFinalizationView',
+
+  //profile settings event
+  // fetchProfileSettings: 'fetchProfileSettings',
+  updateProfileSettings: 'updateProfileSettings',
```
Comment (joshua041904) [inline frontend/source/eventhub/Events.js:31]: This is a good addition.

File: frontend/source/components/ProfileSettingComponent/ProfileSettingComponent.css [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+.slider.round:before {
+    border-radius: 50%;
+}
+
+
+.err-msg {
+    color: #d2171d;
+    font-size: small;
+
+}
```
Comment (wgwumass) [inline frontend/source/components/ProfileSettingComponent/ProfileSettingComponent.css:195]: CSS looks good

**Review by joshua041904 — APPROVED** [review]
Your profile settings implementation looks good. The other changes, like adding a new event to Event.js and including your component in AppControllerComponent, also looks good.

**Pull request merged**